test/promql/tests/test_progressing_argocd.yaml

```diff
@@ -0,0 +1,81 @@
+            exp_annotations:
+              summary: "Application first-test-app in namespace Still-Progressing is in 'Progressing' state for more than 20 minuts"
+              description: "Application first-test-app in namespace Still-Progressing is in 'Progressing' state for more than 20 minuts"
+
+  - interval: 1m
+    input_series:
+      # app was progessing and then Healthy for the last 5 min, should not alert
+      - series: 'argocd_app_info{health_status="Progressing", name="second-test-app", dest_namespace="Progressing-Healthy-5min"}'
+        values: 1x14 0x4
+      - series: 'argocd_app_info{health_status="Healthy", name="second-test-app", dest_namespace="Progressing-Healthy-5min"}'
+        values: 0x14 1x4
+
+  - interval: 1m
+    input_series:
+      # app was progessing and then Healthy for the last 3 min, should alert
+      - series: 'argocd_app_info{health_status="Progressing", name="third-test-app", dest_namespace="Progressing-Healthy-3min"}'
+        values: 1x16 0 0 0
+      - series: 'argocd_app_info{health_status="Healthy", name="third-test-app", dest_namespace="Progressing-Healthy-3min"}'
+        values: 0x16 1 1 1
+    alert_rule_test:
+      - eval_time: 20m
+        alertname: ProgressingArgocdApp
+        exp_alerts:
+          - exp_labels:
+              severity: Critical
+              dest_namespace: Progressing-Healthy-3min
+              health_status: Progressing
+              name: third-test-app
+            exp_annotations:
+              summary: "Application third-test-app in namespace Progressing-Healthy-3min is in 'Progressing' state for more than 20 minuts"
+              description: "Application third-test-app in namespace Progressing-Healthy-3min is in 'Progressing' state for more than 20 minuts"
+
+  - interval: 1m
+    input_series:
+      # app is flapping and alert should be triggered
+      - series: 'argocd_app_info{health_status="Progressing", name="fourth-test-app", dest_namespace="Flapping-Should-Alert"}'
+        values: '1 0 0 1 1 0 1 1 1 0 0 0 0 1 0 1 1 1 1 0'  # 1: 11, 0: 9
+      - series: 'argocd_app_info{health_status="Healthy", name="fourth-test-app", dest_namespace="Flapping-Should-Alert"}'
+        values: '0 1 0 0 0 1 0 0 0 1 1 0 1 0 1 0 0 0 0 0'
+    alert_rule_test:
+      - eval_time: 20m
+        alertname: ProgressingArgocdApp
+        exp_alerts:
+          - exp_labels:
+              severity: Critical
+              dest_namespace: Flapping-Should-Alert
+              health_status: Progressing
+              name: fourth-test-app
+            exp_annotations:
+              summary: "Application fourth-test-app in namespace Flapping-Should-Alert is in 'Progressing' state for more than 20 minuts"
+              description: "Application fourth-test-app in namespace Flapping-Should-Alert is in 'Progressing' state for more than 20 minuts"
+
+  - interval: 1m
+    input_series:
+      # app is flapping and alert should not be triggered
```

Comment:
why does it alert on the former test but not on this one?

Reply:
because this series:
```
- series: 'argocd_app_info{health_status="Progressing", name="fith-test-app", dest_namespace="Flapping-No-Alert"}'
        values: '1 0 0 1 0 0 1 1 1 0 0 0 0 1 0 1 1 1 1 0' 
```
it is `Progressing` 10 out of 20 times in the last 20 minutes, which is `=0.5`, and the rule is for `>0.5`

Reply:
Should be mentioned in the comment, Let's not expect readers to count the ones and zeros...

Reply:
It is no longer relevant since the rule is now changed and no longer consists of an average.